037-auto-rebalance-cluster-scaling.md

```diff
@@ -0,0 +1,177 @@
+4. The cluster operator asks Cruise Control to generate an optimization proposal via the REST API.
+5. When the optimization proposal is ready, the cluster operator checks if the user has specified auto-approval or not:
+   * If auto-approval is not specified, the user has to approve the proposal manually as usual.
+   * If auto-approval and criteria are met or they are not specified at all, the cluster operator approves the proposal automatically by annotating the `KafkaRebalance` custom resource with the `strimzi.io/rebalance=approve` annotation.
+   * If auto-approval and criteria are not met, the `KafkaRebalance` custom resource status will reflect, by using a warning condition, that the proposal was not approved automatically and the reason for that.
+
+It could happen that the cluster was scaling up but rebalancing didn’t happen because the optimization proposal was not approved.
+To fix this kind of issue, the user should change the rebalancing configuration and allow the cluster operator to reconcile again (but not scaling up because it was already done) for getting a new optimization proposal and running a rebalance.
+
+Auto-rebalancing on scaling down.
+
+1. The user scales down the cluster by decreasing the `spec.kafka.replicas` field.
+2. If `spec.kafka.autoRebalance.mode` contains the `scaling` value, the cluster operator creates a `KafkaRebalance` custom resource.
+   * It will specify `spec.mode` as `remove-brokers` and `spec.brokers` with the list of brokers that will be removed.
+3. The cluster operator asks Cruise Control to generate an optimization proposal via the REST API.
+4. When the optimization proposal is ready, the cluster operator checks if the user has specified auto-approval or not:
+   * If auto-approval is not specified, the user has to approve the proposal manually as usual.
+   * If auto-approval and criteria are met or they are not specified at all, the cluster operator approves the proposal automatically by annotating the `KafkaRebalance` custom resource with the `strimzi.io/rebalance=approve` annotation.
+   * If auto-approval and criteria are not met, the `KafkaRebalance` custom resource status will reflect, by using a warning condition, that the proposal was not approved automatically and the reason for that.
+5. Only when the proposal is approved and rebalancing is complete, the cluster operator can continue to scale down the cluster by deleting the brokers.
+
+In this use case, the actual scaling down (deleting the brokers) is blocked on the proposal to be approved and the rebalancing done.
+To fix this kind of issue, the user should change the rebalancing configuration and allow the cluster operator to reconcile again for getting a new optimization proposal, running a rebalance and finally unlocking the scaling down.
+
+## Affected/not affected projects
+
+This proposal impacts the Strimzi Cluster Operator only and mostly the `KafkaRebalanceAssemblyOperator`.
```

Comment:
Can you elaborate? I would say it mostly impacts KafkaReconciler -> I would expect KafkaRebalanceAssemblyOperator based on this proposal to just crunch the `KafkaRebalance` resources without any knowledge about the actor which created them -> user manually or CO during an automated operation.